### ISSUE TITLE
VS build error: cannot use wchar_t string arguments

### ISSUE THREAD

**Issue by LiviuAndron** [gabime/spdlog#851]
**Error	C2440**	`'return': cannot convert from 'void' to 'fmt::v5::internal::value<Context>'`

Microsoft Visual Studio Community 2017, Version 15.8.5
Latest versions of fmt and spdlog, but also with spdlog v1.1.0 and fmt 5.2.1.

```
auto sink_test = std::make_shared<spdlog::sinks::stdout_sink_mt>();
auto logger = std::make_shared<spdlog::logger>("another unicode test", sink_test);
logger->info(L"This is a second test");
logger->info(L"Now for a third test: {}", "Highs and lows");

/*
This build error comes from spdlog:

Error   C2440   'return': cannot convert from 'void' to 'fmt::v5::internal::value<Context>' tests_basic
c:\users\liviu\documents\code\lima\desktop\citron-branches\2-test\tools\vcpkg\installed\x64-windows-citron\include\fmt\core.h   1011
*/
logger->warn(L"This is {} nasty test.", L"another");
```

**Comment by LiviuAndron** [comment]
[minimal_cmake_sample.zip](https://github.com/gabime/spdlog/files/2436952/minimal_cmake_sample.zip)

Workaround:
```
/*
    inefficient workaround: 
    - always compose the string message, no matter the level
    - the UTF-16 string will be converted to UTF-8 (and back again for the Windows API)
*/
const auto msg = fmt::format(L"This is {} nasty test: {}", L"another", 13);
logger->warn(msg.c_str());
```

**Comment by LiviuAndron** [comment]
Related: https://github.com/Microsoft/vcpkg/issues/4384

**Comment by LiviuAndron** [comment]
I use `SPDLOG_FMT_EXTERNAL` and the latest fmt library. Otherwise it is the same as https://github.com/gabime/spdlog/issues/852 .

**Comment by gabime** [comment]
Fixed in 8b42b7d269f3397b04e9e4c8f236c4b922de410b

**Comment by LiviuAndron** [comment]
Thank you!
… and good luck with the vcpkg approval!
